src/plot_api/helpers.js

```diff
@@ -120,17 +119,17 @@ exports.cleanLayout = function(layout) {
             }
             delete ann.ref;
         }
+
         cleanAxRef(ann, 'xref');
         cleanAxRef(ann, 'yref');
     }
 
-    if(layout.shapes !== undefined && !Array.isArray(layout.shapes)) {
-        Lib.warn('Shapes must be an array.');
-        delete layout.shapes;
-    }
     var shapesLen = (layout.shapes || []).length;
```

Comment:
done in https://github.com/plotly/plotly.js/pull/1118/commits/9de2f772e537c7e4fe9cd87458369804ac88a25c